public/pages/CreateIndex/containers/IndexForm/index.tsx

```diff
@@ -105,6 +106,16 @@ export class IndexForm extends Component<IndexFormProps & { services: BrowserSer
       this.refreshIndex();
     }
   }
+  componentDidUpdate(prevProps: IndexFormProps, prevState: CreateIndexState) {
```

Comment:
Should be done in getDerivedStateFromProps method I think.

Reply:
This logic was part of `getDerivedStateFromProps` before. I have refactored it to `componentDidUpdate` as `getDerivedStateFromProps` would have required to keep `dataSourceId` in the state and it was not being used in the component. This component doesn't extend `MDSEnabledComponent`.

Reply:
sorry I did not get it. I guess we can do that by using return 
```
{
  ...prevState,
  ...{
    isSubmitting: false,
    indexDetail: merge({}, defaultIndexSettings, IndexForm.transformIndexDetailToLocal(this.props.value)),
    oldIndexDetail: undefined,
  } 
}
```

Reply:
if dataSourceId is not part of the state, we can't compare if dataSourceId prop has actually changed. as we don't have access to current props (accessible through `this`) in static getDerivedStateFromProps lifecycle method.